openapi3/schema.go

```diff
@@ -50,13 +51,47 @@ func Uint64Ptr(value uint64) *uint64 {
 	return &value
 }
 
+type SchemaMap map[string]*SchemaRef
+
+func (s SchemaMap) JSONLookup(token string) (interface{}, error) {
+	ref, ok := s[token]
+	if ok == false {
+		return nil, fmt.Errorf("object has no field %q", token)
+	}
+
+	if ref.Ref != "" {
+		return &Ref{Ref: ref.Ref}, nil
+	}
+	return ref.Value, nil
+}
+
+type Schemas []*SchemaRef
+
+func (s Schemas) JSONLookup(token string) (interface{}, error) {
+	i, err := strconv.Atoi(token)
+	if err != nil {
+		return nil, err
+	}
+
+	if i >= len(s) {
+		return nil, fmt.Errorf("index out of range: %d", i)
+	}
+
+	ref := s[i]
+
+	if ref.Ref != "" {
+		return &Ref{Ref: ref.Ref}, nil
+	}
+	return ref.Value, nil
+}
+
 // Schema is specified by OpenAPI/Swagger 3.0 standard.
 type Schema struct {
 	ExtensionProps
 
-	OneOf        []*SchemaRef  `json:"oneOf,omitempty" yaml:"oneOf,omitempty"`
-	AnyOf        []*SchemaRef  `json:"anyOf,omitempty" yaml:"anyOf,omitempty"`
-	AllOf        []*SchemaRef  `json:"allOf,omitempty" yaml:"allOf,omitempty"`
+	OneOf        Schemas       `json:"oneOf,omitempty" yaml:"oneOf,omitempty"`
```

Comment:
Not sure if the type Schemas was defined before your PR. 
How about naming these SchemaRefs (slice of SchemaRef pointers) and renaming SchemaMap to Schemas?